enterprise/backend/test/metabase_enterprise/sso/integrations/saml_test.clj

```diff
@@ -669,23 +669,43 @@
                                             :last-name  "user"
                                             :email      "test1234@metabsae.com"})))))))
 
-(deftest logout-should-delete-session-test
-  (testing "Successful SAML SLO logouts should delete the user's session."
-    (let [session-id (str (random-uuid))]
-      (mt/with-temp [:model/User user {:email "saml_test@metabase.com" :sso_source "saml"}
-                     :model/Session _ {:user_id (:id user) :id session-id}]
-        (with-saml-default-setup
-          (is (t2/exists? :model/Session :id session-id))
-          (let [req-options (-> (saml-post-request-options
-                                 (saml-slo-test-response)
-                                 (saml/str->base64 default-redirect-uri))
-                                ;; Client sends their session cookie during the SLO request redirect from the IDP.
-                                (assoc-in [:request-options :cookies mw.session/metabase-session-cookie :value] session-id))
-                response    (client-full-response :post 302 "/auth/sso/handle_slo" req-options)]
-            (is (str/blank? (get-in response [:cookies mw.session/metabase-session-cookie :value]))
-                "After a successful log-out, you don't have a session")
-            (is (not (t2/exists? :model/Session :id session-id))
-                "After a successful log-out, the session is deleted")))))))
+(deftest logout-should-delete-session-test-slo-enabled
+  (testing "Successful SAML SLO logouts should delete the user's session, when saml-slo-enabled."
+    (mt/with-temporary-setting-values [saml-slo-enabled true]
+      (let [session-id (str (random-uuid))]
+        (mt/with-temp [:model/User user {:email "saml_test@metabase.com" :sso_source "saml"}
+                       :model/Session _ {:user_id (:id user) :id session-id}]
+          (with-saml-default-setup
+            (is (t2/exists? :model/Session :id session-id))
+            (let [req-options (-> (saml-post-request-options
+                                   (saml-slo-test-response)
+                                   (saml/str->base64 default-redirect-uri))
+                                  ;; Client sends their session cookie during the SLO request redirect from the IDP.
+                                  (assoc-in [:request-options :cookies mw.session/metabase-session-cookie :value] session-id))
+                  response    (client-full-response :post 302 "/auth/sso/handle_slo" req-options)]
+              (is (str/blank? (get-in response [:cookies mw.session/metabase-session-cookie :value]))
+                  "After a successful log-out, you don't have a session")
+              (is (not (t2/exists? :model/Session :id session-id))
+                  "After a successful log-out, the session is deleted"))))))))
+
+(deftest logout-should-delete-session-test-slo-disabled
```

Comment:
This test should be `logout-should-not-delete-session-test-slo-disabled` (as you've no doubt noticed ;) )

Reply:
fixed

Reply:
🚀